BaSP-a2022/marta-b-trackgenix-app
# BaSP-TG-56: Super admins redux create, update and delete

### What has developed?

**- Create _constants_ to:**
  - create an admin
  - update an admin
  - delete an admin
  - get by id an admin

**- Create _actions_ to:**
  - create an admin
  - update an admin
  - delete an admin
  - get by id an admin

**- Create _thunks_ to:**
  - create an admin
  - update an admin
  - delete an admin
  - get by id an admin

**- Complete superAdmins _reducer_ to:**
  - create an admin
  - update an admin
  - delete an admin
  - get by id an admin

**- Connect _superAdmins component_ with store to:**
  - create an admin
  - update an admin
  - delete an admin
  - get by id an admin

### How to test?:
To test this ticket we should open the link of the app deployed in Vercel. Then, we should verify the correct functionality of:

- In the super admins table, the data has to be retrieved correctly and fill the table with all data Admins.
- In the super admins table, when clicking the delete button of a row, there should be a confirmation modal to delete a super admin. In case of 'No', modal should be closed and in case of 'Yes', it should delete the selected admin.
- When clicking on the edit button inside super admins table row, it should navigate to the super admins form page. We should check that the form gets all the admin info and we should update all the changes we made. Once confirmed all these changes, it should navigate back to the super admins table and it should reflect all changes made in the admin who was selected.
- When clicking on the add button in the super admins table page, it should navigate to the super admins form page. We should complete the form with data and press the confirm button. It should navigate to the super admins table and a new row of admin should appear, with all new data.

The main objective of this test is to verify the correct functionality of the app **_after redux implementation_**.
